docker build出错

按照https://github.com/PaddlePaddle/Paddle/blob/develop/doc/getstarted/build_and_install/docker_install_en.rst 在mac上面进行docker build，出错如下：
In file included from /paddle/paddle/cuda/src/hl_warpctc_wrap.cc:15:0:
/paddle/paddle/cuda/include/hl_warpctc_wrap.h:19:34: fatal error: warp-ctc/include/ctc.h: No such file or directory
 #include "warp-ctc/include/ctc.h"
                                  ^
compilation terminated.
make[2]: *** [paddle/cuda/CMakeFiles/paddle_cuda.dir/src/hl_warpctc_wrap.cc.o] Error 1
make[1]: *** [paddle/cuda/CMakeFiles/paddle_cuda.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

然后发现warp-ctc下的文件为空，重新git clone一遍paddle源代码该文件夹也是空的，是需要自己手动去clone warp-ctc的库么？

--- a/doc/getstarted/build_and_install/docker_install_en.rst
+++ b/doc/getstarted/build_and_install/docker_install_en.rst
@@ -12,7 +12,7 @@ of your hardware resource on Mac OS X and Windows.
 Development Using Docker
 ------------------------
 
-Develpers can work on PaddlePaddle using Docker.  This allows
+Developers can work on PaddlePaddle using Docker.  This allows
 developers to work on different platforms -- Linux, Mac OS X, and
 Windows -- in a consistent way.
 
@@ -22,10 +22,18 @@ The general development workflow with Docker and Bazel is as follows:
 
    .. code-block:: bash
 
-      git clone --recursive https://github.com/paddlepaddle/paddle
+      git clone --recursive https://github.com/PaddlePaddle/Paddle.git
 
    
    Here **git clone --recursive is required** as we have a submodule `warp-ctc <https://github.com/baidu-research/warp-ctc>`_.
+
+   If you have used :code:`git clone https://github.com/PaddlePaddle/Paddle` and find that the directory :code:`warp-ctc` is
+   empty, please use the following command to get the submodule.
+
+   .. code-block:: bash
+
+      git submodule update --init --recursive
+
 
 2. Build a development Docker image :code:`paddle:dev` from the source
    code.  This image contains all the development tools and
@@ -154,7 +162,7 @@ source code:
 .. code-block:: bash
 
    cd ~
-   git clone github.com/PaddlePaddle/Paddle
+   git clone https://github.com/PaddlePaddle/Paddle.git
    cd Paddle
    git submodule update --init --recursive
    docker build --build-arg WITH_AVX=OFF -t paddle:cpu-noavx -f paddle/scripts/docker/Dockerfile .
@@ -170,7 +178,7 @@ generated using `woboq code browser
 for users to browse and understand the C++ source code.
 
 As long as we give the Paddle Docker container a name, we can run an
-additional nginx Docker container to serve the volume from the Paddle
+additional Nginx Docker container to serve the volume from the Paddle
 container:
 
 .. code-block:: bash

--- a/doc/getstarted/build_and_install/docker_install_en.rst
+++ b/doc/getstarted/build_and_install/docker_install_en.rst
@@ -9,6 +9,83 @@ Please be aware that you will need to change `Dockers settings
 of your hardware resource on Mac OS X and Windows.
 
 
+Development Using Docker
+------------------------
+
+Develpers can work on PaddlePaddle using Docker.  This allows
+developers to work on different platforms -- Linux, Mac OS X, and
+Windows -- in a consistent way.
+
+The general development workflow with Docker and Bazel is as follows:
+
+1. Get the source code of Paddle:
+
+   .. code-block:: bash
+
+      git clone --recursive https://github.com/paddlepaddle/paddle
+
+   
+   Here **git clone --recursive is required** as we have a submodule `warp-ctc <https://github.com/baidu-research/warp-ctc>`_.
+
+2. Build a development Docker image :code:`paddle:dev` from the source
+   code.  This image contains all the development tools and
+   dependencies of PaddlePaddle.
+
+
+   .. code-block:: bash
+
+      cd paddle
+      docker build -t paddle:dev -f paddle/scripts/docker/Dockerfile .
+
+
+3. Run the image as a container and mounting local source code
+   directory into the container.  This allows us to change the code on
+   the host and build it within the container.
+
+   .. code-block:: bash
+
+      docker run       \
+       -d              \
+       --name paddle   \
+       -p 2022:22      \
+       -v $PWD:/paddle \
+       -v $HOME/.cache/bazel:/root/.cache/bazel \
+       paddle:dev
+
+   where :code:`-d` makes the container running in background,
+   :code:`--name paddle` allows us to run a nginx container to serve
+   documents in this container, :code:`-p 2022:22` allows us to SSH
+   into this container, :code:`-v $PWD:/paddle` shares the source code
+   on the host with the container, :code:`-v
+   $HOME/.cache/bazel:/root/.cache/bazel` shares Bazel cache on the
+   host with the container.
+
+4. SSH into the container:
+
+   .. code-block:: bash
+
+      ssh root@localhost -p 2022
+
+5. We can edit the source code in the container or on this host.  Then
+   we can build using cmake
+
+   .. code-block:: bash
+
+      cd /paddle # where paddle source code has been mounted into the container
+      mkdir -p build
+      cd build
+      cmake -DWITH_TESTING=ON ..
+      make -j `nproc`
+      CTEST_OUTPUT_ON_FAILURE=1 ctest
+
+   or Bazel in the container:
+
+   .. code-block:: bash
+
+      cd /paddle
+      bazel test ...
+
+
 CPU-only and GPU Images
 -----------------------
 
@@ -104,78 +181,3 @@ container:
 
 Then we can direct our Web browser to the HTML version of source code
 at http://localhost:8088/paddle/
-
-
-Development Using Docker
-------------------------
-
-Develpers can work on PaddlePaddle using Docker.  This allows
-developers to work on different platforms -- Linux, Mac OS X, and
-Windows -- in a consistent way.
-
-The general development workflow with Docker and Bazel is as follows:
-
-1. Get the source code of Paddle:
-
-   .. code-block:: bash
-
-      git clone --recursive https://github.com/paddlepaddle/paddle
-
-
-2. Build a development Docker image :code:`paddle:dev` from the source
-   code.  This image contains all the development tools and
-   dependencies of PaddlePaddle.
-
-
-   .. code-block:: bash
-
-      cd paddle
-      docker build -t paddle:dev -f paddle/scripts/docker/Dockerfile .
-
-
-3. Run the image as a container and mounting local source code
-   directory into the container.  This allows us to change the code on
-   the host and build it within the container.
-
-   .. code-block:: bash
-
-      docker run       \
-       -d              \
-       --name paddle   \
-       -p 2022:22      \
-       -v $PWD:/paddle \
-       -v $HOME/.cache/bazel:/root/.cache/bazel \
-       paddle:dev
-
-   where :code:`-d` makes the container running in background,
-   :code:`--name paddle` allows us to run a nginx container to serve
-   documents in this container, :code:`-p 2022:22` allows us to SSH
-   into this container, :code:`-v $PWD:/paddle` shares the source code
-   on the host with the container, :code:`-v
-   $HOME/.cache/bazel:/root/.cache/bazel` shares Bazel cache on the
-   host with the container.
-
-4. SSH into the container:
-
-   .. code-block:: bash
-
-      ssh root@localhost -p 2022
-
-5. We can edit the source code in the container or on this host.  Then
-   we can build using cmake
-
-   .. code-block:: bash
-
-      cd /paddle # where paddle source code has been mounted into the container
-      mkdir -p build
-      cd build
-      cmake -DWITH_TESTING=ON ..
-      make -j `nproc`
-      CTEST_OUTPUT_ON_FAILURE=1 ctest
-
-   or Bazel in the container:
-
-   .. code-block:: bash
-
-      cd /paddle
-      bazel test ...

